packages/utils/src/dal/mikro-orm/integration-tests/__tests__/mikro-orm-repository.spec.ts

```diff
@@ -0,0 +1,844 @@
+      expect(listedEntity1[0].entity3.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en3-1",
+          }),
+          expect.objectContaining({
+            title: "en3-2",
+          }),
+        ])
+      )
+
+      expect(listedEntity3).toHaveLength(1)
+      expect(listedEntity3[0]).toEqual(
+        expect.objectContaining({
+          title: "en3-1",
+        })
+      )
+      expect(listedEntity3[0].entity1.getItems()).toHaveLength(1)
+      expect(listedEntity3[0].entity1.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en1",
+          }),
+        ])
+      )
+    })
+
+    it("should only update the parent entity of a many-to-many if relation is not included", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity3: [{ title: "en3-1" }, { title: "en3-2" }],
+      }
+
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity3"],
+        relationsToSkip: [],
+      })
+      entity1.title = "newen1"
+      entity1.entity3.push({ title: "en3-3" })
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: [],
+        relationsToSkip: ["entity3"],
+      })
+
+      const listedEntities = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity3"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0].title).toEqual("newen1")
+
+      expect(listedEntities[0].entity3.getItems()).toHaveLength(2)
+      expect(listedEntities[0].entity3.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en3-1",
+          }),
+          expect.objectContaining({
+            title: "en3-2",
+          }),
+        ])
+      )
+    })
+
+    it("should successfully update, create, and delete subentities an entity with a many-to-many relation", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity3: [
+          { id: "4", title: "en3-1" },
+          { id: "5", title: "en3-2" },
+        ] as any,
+      }
+
+      let resp = await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity3"],
+        relationsToSkip: [],
+      })
+      entity1.title = "newen1"
+      entity1.entity3 = [{ id: "4", title: "newen3-1" }, { title: "en3-4" }]
+      resp = await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity3"],
+        relationsToSkip: [],
+      })
+
+      const listedEntities = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity3"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0].title).toEqual("newen1")
+      expect(listedEntities[0].entity3.getItems()).toHaveLength(2)
```

Comment:
sugg: assert that only the pivot tables are being deleted and keeping the inverse entities alive.